src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala

```diff
@@ -452,21 +452,53 @@ final case class AgreementApiServiceImpl(
           .contains(state)
       )
 
+    def getUpdateSeed(
+      newState: AgreementManagement.AgreementState,
+      agreement: AgreementManagement.Agreement,
+      stamps: Stamps
+    ): UpdateAgreementSeed =
+      if (newState == AgreementManagement.AgreementState.ACTIVE)
+        AgreementManagement.UpdateAgreementSeed(
+          state = newState,
+          certifiedAttributes = matchingCertifiedAttributes(eService, consumer),
+          declaredAttributes = matchingDeclaredAttributes(eService, consumer),
+          verifiedAttributes = matchingVerifiedAttributes(eService, consumer),
+          suspendedByConsumer = agreement.suspendedByConsumer,
+          suspendedByProducer = agreement.suspendedByProducer,
+          suspendedByPlatform = suspendedByPlatform,
+          stamps = stamps
+        )
+      else
+        AgreementManagement.UpdateAgreementSeed(
+          state = newState,
+          certifiedAttributes = Nil,
+          declaredAttributes = Nil,
+          verifiedAttributes = Nil,
+          suspendedByConsumer = None,
+          suspendedByProducer = None,
+          suspendedByPlatform = suspendedByPlatform,
+          stamps = stamps
+        )
+
+    def performActivation(agreement: AgreementManagement.Agreement, seed: UpdateAgreementSeed) =
+      for {
+        _ <- agreementContractCreator.create(agreement, eService, consumer, seed)
+        _ <- authorizationManagementService
+          .updateStateOnClients(
+            eServiceId = agreement.eserviceId,
+            consumerId = agreement.consumerId,
+            agreementId = agreement.id,
+            state = toClientState(newState)
+          )
+      } yield ()
```

Comment:
Readability stuff: how about not using a for comprehension and concatenating the 2 ops with `>>`?